src-tauri/src/main.rs

```diff
@@ -72,79 +71,90 @@ async fn control_grabbing(
     frontend_bool: bool,
 ) -> Result<(), ()> {
     state.set_is_listening(frontend_bool);
-    // set_macro_listening(state, frontend_bool);
+    Ok(())
+}
+
+/// Enables or disables the automatic startup of Wootomation at system start.
+fn init_autostart(app_name: &str, set_autolaunch: bool, current_exe: PathBuf) -> Result<(), Error> {
+    let auto_start = auto_launch::AutoLaunchBuilder::new()
+        .set_app_name(app_name)
+        .set_app_path(current_exe.as_path().to_str().unwrap())
+        .set_use_launch_agent(true)
+        .build()
+        .context("App name is empty, or unsupported OS is used.")?;
+
+    match set_autolaunch {
+        true => auto_start.enable()?,
+        false => {
+            if let Err(e) = auto_start.disable() {
+                match e {
+                    auto_launch::Error::Io(err) => match err.kind() {
+                        std::io::ErrorKind::NotFound => {
+                            trace!("Autostart is already removed, finished checking.");
+                        }
+                        _ => return Err(err.into()),
+                    },
+                    _ => return Err(e.into()),
+                }
```

Comment:
Are you able to narrow to the `std::io::ErrorKind::NotFound` in the first branch on e? Then you won't need the double `return Err(err.into())`

Reply:
reverted to original after discussion